internal/pkg/template/templates/workloads/partials/cf/nlb.yml

```
@@ -0,0 +1,111 @@
+PublicNetworkLoadBalancer:
+  Metadata:
+    'aws:copilot:description': 'A Network Load Balancer to distribute public traffic to your service'
+  Condition: CreateNLB # TODO: Add a condition in service template.
+  Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+  Properties:
+    IpAddressType: ipv4
+    Scheme: internet-facing
+    Subnets:
+      Fn::importValue:
+        !Sub ${AppName}-${EnvName}-PublicSubnets
+    Type: String
+    # Name: omitted. Let CFN generates one for us.
+    # SubnetMappings: omitted. Specify `Subnets` instead. `SubnetMappings` is used when we need a static Elastic IP address to be associated with a subnet.
+    # Tags: omitted. The template has app/env/service tags, which should be propagated to this lb resource.
+    # LoadBalancerAttributes: omitted.
+    #    - The three available attributes for an NLB are: deletion_protection, access_logs, cross_zone load balancing. All are disabled by default.
+
+NLBListener:
+  Type: AWS::ElasticLoadBalancingV2::Listener
+  Properties:
+    DefaultActions:
+      - TargetGroupArn: !Ref NLBTargetGroup
+        Type: forward
+    LoadBalancerArn: !Ref PublicNetworkLoadBalancer
+    Port: {{ .ListenerPort }}
+    Protocol: {{ .Protocol }}
```

I wonder if we should group these two under the same struct? 💭 
```suggestion
    Port: {{ .NLB.Listener.Port }}
    Protocol: {{ .NLB.Listener.Protocol }}
```